### PR TITLE
Add OpenStack platform to list of allowed dual-stack clusters

### PR DESCRIPTION
**Pull request by MaysaMacedo**
Partially implements: https://github.com/openshift/enhancements/pull/1365

### CODE REVIEW

**Comment by MaysaMacedo** [comment]
/hold
Dual-stack support needs to merge on installer first

**Review by dulek — CHANGES_REQUESTED** [review]

File: pkg/network/render.go [hunk 1]
```diff
@@ -27,6 +27,7 @@ var dualStackPlatforms = sets.NewString(
 	string(configv1.BareMetalPlatformType),
 	string(configv1.NonePlatformType),
 	string(configv1.VSpherePlatformType),
+	string(configv1.OpenStackPlatformType),
```
[the remark this answers is not in the document]
Comment (MaysaMacedo) [reply pkg/network/render.go:30]: done. Thanks!

@dulek Can you take another look?

File: pkg/network/render.go [hunk 2]
```diff
@@ -353,6 +354,8 @@ func isNetworkChangeSafe(prev, next *operv1.NetworkSpec, infraRes *bootstrap.Inf
 		if !isSupportedDualStackPlatform(infraRes.PlatformType) {
 			return errors.Errorf("%s is not one of the supported platforms for dual stack (%s)", infraRes.PlatformType,
 				strings.Join(dualStackPlatforms.List(), ", "))
+		} else if string(configv1.OpenStackPlatformType) == string(infraRes.PlatformType) {
```
Comment (dulek) [inline pkg/network/render.go:357]: We could probably just split `isSupportedDualStackPlatform()` into another function to be used here, but no big deal if it's just OpenStack that doesn't support migrations.

**Review by dulek — COMMENTED** [review]
/lgtm

**Comment by MaysaMacedo** [comment]
/cc @jcaamano 

can you have a look on this PR?

**Comment by MaysaMacedo** [comment]
I plan to remove the hold as soon as https://github.com/openshift/installer/pull/6797 gets merged

**Comment by dcbw** [comment]
We'd probably need a Docs PR then to note that OSP isn't supported for conversion, unless that's getting added very soon?

**Comment by dcbw** [comment]
Also (hopefully??) we can do a dualstack OSP CI job?

**Comment by MaysaMacedo** [comment]
@dcbw we're targeting 4.15 to GA dual-stack, which will include the support for conversion from IPv4 single stack to dual-stack clusters. On 4.14 dual-stack support will be tech preview.

I'm adding [a note in the installer](https://github.com/openshift/installer/pull/7269/files#diff-aa965cd93a12a9ef8cd6b3b94c905e5be363422f096177b20b251a0433fd3222R16) docs about the no support for conversion and will also add to the openshift-docs.

We're working on the CI at the moment https://issues.redhat.com/browse/OSASINFRA-2251.

**Pull request merged**